Add publish and release as separate actions (#106)
bakdata/ci-templates: .github/workflows/README.md

- [Java Gradle Library](https://github.com/bakdata/ci-templates/tree/main/.github/workflows#java-gradle-library)
- [Java Gradle Plugin](https://github.com/bakdata/ci-templates/tree/main/.github/workflows#java-gradle-plugin)
- [Java Gradle Release](https://github.com/bakdata/ci-templates/tree/main/.github/workflows#java-gradle-release)

> - [Python Poetry Publish](https://github.com/bakdata/ci-templates/tree/main/.github/workflows#java-gradle-release)

Reviewer: please put Poetry publish directly underneath Poetry release

## Python Poetry Release

> This workflow will bump the version of your python project, tag and make a release of your project on GitHub. Moreover
> this workflow allows you to add a CHANGELOG.md automatically if you wish to do so.
> In the following, you will first find the necessary prerequisite to set up the workflow. Next, you will find the

Reviewer: ```suggestion
This workflow will bump the version of your Python project, create a Git tag, and make a release of your project on GitHub. Moreover, this workflow allows you to add a CHANGELOG.md automatically if you wish to do so.
In the following, you will first find the necessary prerequisites to set up the workflow. Next, you will find the
```

    with:
      release-type: patch # (Required) See more values at: https://python-poetry.org/docs/cli/#version
      ref: my-awesome-ref # (Optional) if not set the ${{ github.event.repository.default_branch }} will fill the value. In this case the changes will be pushed to my-awesome-ref
      python-version: 3.8 # (Optional) Default value is 3.10. In this case Poetry is installed with Python 3.8

>       poetry-version: 1.5.1 # (Optional) Default value is 1.5.1. In this case Poetry version 1.1.11 is installed

Reviewer: value should be left for the example stated in the comment 
```suggestion
      poetry-version: "1.1.11" # (Optional) Default value is 1.5.1. In this case Poetry version 1.1.11 is installed
```

### Secret Parameters

These secrets define the pypi token that allow the GitHub action to release the project to PyPI or TestPyPI

| Name       | Required | Description                                      |
| ---------- | :------: | ------------------------------------------------ |
| pypi-token |    ✅    | The (Test)PyPI API token for publishing packages |

### Calling the workflow

```yaml
name: Call this reusable workflow

on:
  push:
    tags: 
      - "*"

jobs:
  call-workflow-passing-data:
    uses: bakdata/ci-templates/.github/workflows/python-poetry-publish.yaml@main
    with:
      ref: my-awesome-ref # (Optional) if not set the ${{ github.event.repository.default_branch }} will fill the value. In this case the changes will be pushed to my-awesome-ref
      publish-to-test: false # (Optional) By default the packages are published to TestPyPI. In this case the packages are published to PyPI
      python-version: 3.8 # (Optional) Default value is 3.10. In this case Poetry is installed with Python 3.8

>       poetry-version: 1.5.1 # (Optional) Default value is 1.5.1. In this case Poetry version 1.1.11 is installed

Reviewer: same as above
```suggestion
      poetry-version: "1.1.11" # (Optional) Default value is 1.5.1. In this case Poetry version 1.1.11 is installed
```
